File: GFG/Python/Easy/minimum-depth-of-a-binary-tree.py
# https://www.geeksforgeeks.org/problems/minimum-depth-of-a-binary-tree/1

class Solution:
    def min_depth(self, root):
        #Code here
        if root == None:
            return 0
            
        depth = 0
        queue = [root]
        
        while len(queue):
            depth += 1
            loop_length = len(queue)
            for _ in range(0, loop_length):
                node = queue.pop(0)
                if node.left == None and node.right == None:
                    return depth
                    
                if node.left:
                    queue.append(node.left)
                    
                if node.right:
                    queue.append(node.right)
        return depth

File: GFG/JavaScript/DSA/Easy/minimum-depth-of-a-binary-tree.js
// https://www.geeksforgeeks.org/problems/minimum-depth-of-a-binary-tree/1

class Solution {
  minDepth(root) {
    if (!root) {
      return 0;
    }
    let depth = 0;
    let queue = [root];
    while (queue.length > 0) {
      depth++;
      let levelNodes = queue.length;
      for (let i = 0; i < levelNodes; i++) {
        let node = queue.shift();

        if (node.left == null && node.right == null) {
          return depth;
        }
        if (node.left != null) {
          queue.push(node.left);
        }
        if (node.right != null) {
          queue.push(node.right);
        }
      }
    }
    return depth;
  }
}
